File: src/main/java/com/example/quotebook/model/Quote.java
package com.example.quotebook.model;

import java.time.Instant;

import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "quotes")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Quote {

    @Id
    private String id;

    private String text;
    private String author;

    @CreatedDate
    private Instant createdAt;
    @LastModifiedDate
    private Instant modifiedAt;

    private Integer likes = 0;
}

File: frontend/src/model/quote.ts
export interface Link {
    rel: string
    href: string
}

export interface QuoteModel {
    id: string
    text: string
    author: string
    likes: number
    createdAt: string
    modifiedAt: string
    links: Link[]
}

export interface QuoteRequestModel {
    text: string
    author: string
}